eslint.config.mjs

```diff
@@ -0,0 +1,366 @@
+            '@typescript-eslint/no-redundant-type-constituents': 'off',
+            '@typescript-eslint/no-unsafe-member-access': 'off',
+            '@typescript-eslint/no-unsafe-call': 'off',
+            '@typescript-eslint/no-unsafe-assignment': 'off',
+            '@typescript-eslint/no-unsafe-return': 'off',
+            '@typescript-eslint/no-unsafe-argument': 'off',
+            '@typescript-eslint/no-unnecessary-type-assertion': 'off',
+        },
+    },
+    {
+        files: [
+            '**/__tests__/**',
+            'packages/@lwc/*/scripts/**',
+            'packages/@lwc/synthetic-shadow/index.js',
+        ],
+
+        languageOptions: {
+            ecmaVersion: 5,
+            sourceType: 'script',
+
+            parserOptions: {
+                project: './tsconfig.eslint.json',
+            },
+        },
+    },
+    {
+        files: ['**/*.js', '**/*.mjs', '**/*.cjs'],
+        ...tseslint.configs.disableTypeChecked,
+    },
+    {
+        files: ['commitlint.config.js', '**/jest.config.cjs'],
+        languageOptions: {
+            globals: {
+                ...globals.node,
+            },
+        },
+    },
+    {
+        files: [
+            '**/scripts/**',
+            '**/jest.config.js',
+            'packages/@lwc/integration-tests/src/components/**/*.spec.js',
+        ],
+
+        rules: {
+            '@typescript-eslint/no-var-requires': 'off',
+        },
+    },
+    {
+        files: ['packages/lwc/**'],
+
+        rules: {
+            'no-restricted-imports': 'off',
+        },
+    },
+    {
+        files: [
+            'packages/@lwc/engine-core/**',
+            'packages/@lwc/engine-dom/**',
+            'packages/@lwc/synthetic-shadow/**',
+        ],
+
+        rules: {
+            '@lwc/lwc-internal/no-global-node': 'error',
+            'prefer-rest-params': 'off',
+            'prefer-spread': 'off',
+        },
+    },
+    {
+        files: ['**/__tests__/**', '**/__mocks__/**', 'packages/@lwc/integration-karma/**'],
+
+        languageOptions: {
+            globals: {
+                ...globals.jest,
+                ...globals.es2021,
+            },
+        },
+
+        plugins: {
+            jest,
+        },
+
+        rules: {
+            'jest/no-focused-tests': 'error',
+            'jest/valid-expect': 'error',
+            'jest/valid-expect-in-promise': 'error',
+        },
+    },
+    {
+        files: ['packages/@lwc/integration-tests/**'],
+
+        languageOptions: {
+            globals: {
+                $: true,
+                browser: true,
+                ...globals.browser,
+                ...globals.mocha,
+                ...globals.node,
+            },
+        },
+    },
+    {
+        files: ['./*.js', '**/scripts/**', '**/jest.config.js'],
+
+        languageOptions: {
+            globals: {
+                ...globals.node,
+                ...globals.jest,
+            },
+        },
+
```

Comment:
```suggestion
```
There's a lot of rules going on here, but I _think_ this is covered by other cases?

Reply:
consolidated.